src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -15,103 +15,129 @@ namespace System.Net.Http
     /// </summary>
     internal sealed class DiagnosticsHandler : DelegatingHandler
     {
-        /// <summary>
-        /// DiagnosticHandler constructor
-        /// </summary>
-        /// <param name="innerHandler">Inner handler: Windows or Unix implementation of HttpMessageHandler.
-        /// Note that DiagnosticHandler is the latest in the pipeline </param>
-        public DiagnosticsHandler(HttpMessageHandler innerHandler) : base(innerHandler)
-        {
-        }
+        private const string Namespace                      = "System.Net.Http";
+        private const string RequestWriteNameDeprecated     = Namespace + ".Request";
+        private const string ResponseWriteNameDeprecated    = Namespace + ".Response";
+        private const string ExceptionEventName             = Namespace + ".Exception";
+        private const string ActivityName                   = Namespace + ".HttpRequestOut";
+        private const string ActivityStartName              = ActivityName + ".Start";
+        private const string ActivityStopName               = ActivityName + ".Stop";
 
-        internal static bool IsEnabled()
+        private static readonly DiagnosticListener s_diagnosticListener = new("HttpHandlerDiagnosticListener");
+        private static readonly ActivitySource s_activitySource = new(Namespace);
+
+        public static readonly bool IsGloballyEnabled = GetEnableActivityPropagationValue();
```

Comment:
>IsGloballyEnabled [](http://example.com/codeflow?start=36&length=17)

s_isGloballyEnabled?

Reply:
It's now a `public` static, AFAIK we only use the `s_` for private ones

Reply:
Isn't the class still internal? so technically it is not public. If we expose this class later publicly, we wouldn't expose this as static field. instead, we'll do it as a property.

Reply:
Changed this to a property now due to https://github.com/dotnet/runtime/pull/54437#discussion_r656056013